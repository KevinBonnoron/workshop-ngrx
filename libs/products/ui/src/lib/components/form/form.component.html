<form [formGroup]="formGroup">
  <mat-form-field appearance="outline">
    <mat-label>Name</mat-label>
    <input matInput formControlName="name" />
  </mat-form-field>

  <button mat-raised-button color="primary" (click)="save.emit(formGroup.getRawValue())">Valider</button>
</form>
