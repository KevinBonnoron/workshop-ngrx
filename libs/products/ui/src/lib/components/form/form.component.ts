import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges, inject } from '@angular/core';
import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
import { MatButtonModule } from '@angular/material/button';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatInputModule } from '@angular/material/input';
import { Product } from '@workshop/shared';

@Component({
  standalone: true,
  selector: 'products-ui-form',
  imports: [ReactiveFormsModule, MatButtonModule, MatFormFieldModule, MatInputModule],
  templateUrl: './form.component.html'
})
export class FormComponent implements OnChanges {
  private readonly formBuilder = inject(FormBuilder);

  readonly formGroup = this.formBuilder.nonNullable.group<Product>({
    id: 0,
    name: ''
  });

  @Input()
  product!: Product;

  @Output()
  readonly save = new EventEmitter<Product>();

  ngOnChanges(changes: SimpleChanges): void {
    if (changes['product']) {
      this.formGroup.patchValue(this.product);
    }
  }
}
