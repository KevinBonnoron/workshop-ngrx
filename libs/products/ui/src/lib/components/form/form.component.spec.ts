import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FormComponent } from './form.component';

describe('FormComponent', () => {
  let fixture: ComponentFixture<FormComponent>;
  let component: FormComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [FormComponent],
    });

    fixture = TestBed.createComponent(FormComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snaptshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
