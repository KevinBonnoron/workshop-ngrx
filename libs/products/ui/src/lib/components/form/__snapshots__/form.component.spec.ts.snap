// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormComponent should match snaptshot 1`] = `
<products-ui-form
  formBuilder={[Function _FormBuilder]}
  formGroup={[Function FormGroup]}
  save={[Function EventEmitter_]}
>
  <form
    novalidate=""
  >
    <mat-form-field
      appearance="outline"
      class="mat-mdc-form-field"
    >
      <div
        class="mat-mdc-text-field-wrapper mdc-text-field"
      >
        <div
          class="mat-mdc-form-field-flex"
        >
          <div
            class="mat-mdc-form-field-infix"
          >
            <input
              class="mat-mdc-input-element"
              formcontrolname="name"
              matinput=""
            />
          </div>
        </div>
      </div>
      <div
        class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
      >
      </div>
    </mat-form-field>
    <button
      class="mdc-button mdc-button--raised mat-mdc-raised-button mat-primary"
      color="primary"
      mat-raised-button=""
      mat-ripple-loader-class-name="mat-mdc-button-ripple"
      mat-ripple-loader-uninitialized=""
    >
      <span
        class="mat-mdc-button-persistent-ripple"
      />
      <span
        class="mdc-button__label"
      >
        Valider
      </span>
      <span
        class="mat-mdc-focus-indicator"
      />
      <span
        class="mat-mdc-button-touch-target"
      />
    </button>
  </form>
</products-ui-form>
`;
